client/src/Components/Appointments/AppointmentFormModal.js

```diff
@@ -142,6 +145,86 @@ export default function AppointmentFormModal(props) {
     });
   };
 
+  // Appointment validation function
+  const handleSubmitModal = (e, appointment, patient) => {
+    e.preventDefault();
+
+    // New appointment time variables (start time and end time)
+    const startTime = moment(`${appointment.date} ${appointment.start}`).unix();
+    const endTime = moment(`${appointment.date} ${appointment.end}`).unix();
+
+    /* First way
+     * Find overlapped time (using for in loop)
+     * Return setErrors(errorMessage) if there is overlap
+     * Or
+     * Call onClick function if there is no overlap
+     * =========Time Complexity=========
+     * for in loop: worst O(n)
+     * ======> Total time complexity: worst O(n)
+     */
+    for (let e in events) {
+      if (
+        moment(events[e].start).unix() < startTime &&
+        startTime < moment(events[e].end).unix()
+      ) {
+        return setErrors(
```

Comment:
You don't have to return `setErrors` since it will always setErrors without returning it.  Also Functions doesn't always need to return value.

Reply:
To escape the function, I used return setErrors. Without it, It calls parent function. Do I need to change syntax to escape the function without return?

Reply:
What you can do here is create `let validationError;` before `for loop` then assign error message to `validationError` = `error message` inside `if` `else if` statements.  Then `if (validationError)` is exists, `setError(validationError)`, else `onClick(e, appointment, patient)`

Reply:
You can also shorten if check by combining both conditions into one since they both return same exact error message.